### ISSUE TITLE
Overlay isn't shown despite TypeScript errors

### ISSUE THREAD

**Issue by ulrichstark** [fi3ework/vite-plugin-checker#733]
### Describe the bug

Despite my project having TypeScript errors, the enabled overlay of `vite-plugin-checker` doesn't show the overlay in the dev-served web application. The TS error is reported in the terminal by `vite-plugin-checker`.

### Reproduction

```sh
git clone https://github.com/ulrichstark/repro-vite-plugin-checker-overlay.git
cd repro-vite-plugin-checker-overlay
npm i
npx vite
```

### Expected behavior

The overlay should be visible in the dev-served web application if there are TS errors and the overlay is enabled in  `vite.config.ts`.

### System Info

```shell
System:
    OS: macOS 26.3
    CPU: (10) arm64 Apple M4
    Memory: 141.44 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 26.1.0 - /Users/ulrichstark/.nvm/versions/node/v26.1.0/bin/node
    npm: 11.13.0 - /Users/ulrichstark/.nvm/versions/node/v26.1.0/bin/npm
    pnpm: 10.12.2 - /opt/homebrew/bin/pnpm
    bun: 1.2.16 - /Users/ulrichstark/.bun/bin/bun
    Deno: 2.7.12 - /Users/ulrichstark/.deno/bin/deno
  Browsers:
    Chrome: 148.0.7778.169
    Firefox: 151.0.2
    Safari: 26.3
  npmPackages:
    vite-plugin-checker: ^0.14.1 => 0.14.1
```

### Validations

- [x] Read the [docs](https://github.com/fi3ework/vite-plugin-checker#readme).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by ulrichstark** [comment]
## AI diagnosis of my issue

**The overlay never renders because its client script is never injected into TanStack Start's HTML.**

vite-plugin-checker delivers its overlay in two halves:

1. **Server side** — `configureServer` sends errors over the Vite WebSocket: `server.ws.send("vite-plugin-checker", payload)`. ✅ This works — the terminal shows the error is detected.
2. **Client side** — `transformIndexHtml` injects a `<script type="module">` preamble that imports the runtime, which subscribes to that WS channel and draws the overlay.

The problem: **`transformIndexHtml` only runs on HTML that passes through Vite's index-html pipeline.** TanStack Start renders the page shell itself via React SSR (`__root.tsx`'s `shellComponent`) and serves it directly, bypassing that hook entirely.

I confirmed this by curling the dev server. The served HTML contains TanStack's own client entry but **zero** reference to vite-plugin-checker:

```html
<head>...<link rel="modulepreload" href="/@id/virtual:tanstack-start-dev-client-entry"/></head>
```

So the WS messages are broadcast correctly, but nothing in the browser is listening → no overlay. Terminal reporting is unaffected, exactly matching the report.

## Workaround (verified)

The plugin still registers the runtime as a virtual module, reachable at `/@id/virtual:@vite-plugin-checker-runtime-entry`. Manually loading it from the shell head in dev restores the overlay (confirmed it appears in the served HTML):

```tsx
<head>
    <HeadContent />
    {import.meta.env.DEV && (
        <script type="module" src="/@id/virtual:@vite-plugin-checker-runtime-entry" />
    )}
</head>
```

## Real fix (plugin side)

The plugin shouldn't rely solely on `transformIndexHtml`. Options:

- Inject the preamble via a `transform` hook on the SSR/client entry module (works regardless of who renders the HTML), or
- Document the virtual-module import as the supported path for SSR frameworks like TanStack Start, Nuxt, SvelteKit, etc.
